core/domain/story_services.py

```diff
@@ -241,6 +246,226 @@ def save_new_story(committer_id, story):
+    successful, increments the version number of the incoming story domain
+    object by 1.
+
+    Args:
+        committer_id: str. ID of the given committer.
+        story: Story. The story domain object to be saved.
+        commit_message: str. The commit message.
+        change_list: list(dict). List of changes applied to a story. Each
+            entry in change_list is a dict that represents a StoryChange.
+
+    Raises:
+        ValidationError: An invalid exploration was referenced in the
+            story.
+        Exception: The story model and the incoming story domain
+            object have different version numbers.
+    """
+    if not change_list:
+        raise Exception(
+            'Unexpected error: received an invalid change list when trying to '
+            'save story %s: %s' % (story.id, change_list))
+
+    # Validate that all explorations referenced by the story exist.
+    exp_ids = []
+    for node in story.story_contents.nodes:
+        exp_ids.append(node.exploration_id)
+    exp_summaries = (
+        exp_services.get_exploration_summaries_matching_ids(exp_ids))
+    exp_summaries_dict = {
+        exp_id: exp_summaries[ind] for (ind, exp_id) in enumerate(exp_ids)
+    }
+    for node in story.story_contents.nodes:
+        if (node.exploration_id is not None) and (
+                not exp_summaries_dict[node.exploration_id]):
+            raise utils.ValidationError(
+                'Expected story to only reference valid explorations, '
+                'but found an exploration with ID: %s (was it deleted?)' %
+                node.exploration_id)
+
+    story_model = story_models.StoryModel.get(story.id, strict=False)
+    if story_model is None:
+        story_model = story_models.StoryModel(id=story.id)
+    else:
+        if story.version > story_model.version:
+            raise Exception(
+                'Unexpected error: trying to update version %s of story '
+                'from version %s. Please reload the page and try again.'
+                % (story_model.version, story.version))
+        elif story.version < story_model.version:
+            raise Exception(
+                'Trying to update version %s of story from version %s, '
+                'which is too old. Please reload the page and try again.'
+                % (story_model.version, story.version))
+
+    story_model.description = story.description
+    story_model.title = story.title
+    story_model.notes = story.notes
+    story_model.language_code = story.language_code
+    story_model.schema_version = story.schema_version
+    story_model.story_contents = {
+        'nodes': [
+            node.to_dict() for node in story.story_contents.nodes
+        ]
+    }
+    story_model.version = story.version
+    story_model.commit(committer_id, commit_message, change_list)
+    memcache_services.delete(_get_story_memcache_key(story.id))
+    index_story_given_id(story.id)
```

Comment:
Probably not needed, as well as all the other search index stuff.

Reply:
Ok, it was that initially, I only did the commit line, but then the tests were failing. So I wrote both these lines but it was because I did not have the memcache_services.delete() line before. Will remove this one.

Reply:
Done.